packages/Webkul/Checkout/src/Models/Cart.php

```diff
@@ -123,13 +123,29 @@ public function haveStockableItems()
     public function hasDownloadableItems()
     {
         foreach ($this->items as $item) {
-            if ($item->type == 'downloadable')
+            if (stristr($item->type,'downloadable') !== false) {
                 return true;
+            }
         }
 
         return false;
     }
 
+    /**
+     * Returns true if cart contains one or many products with quantity box.
+     * f.e. simple, configurable, virtual
```

Comment:
"f.e.": Did you mean "for example"? The abbreviation for that would be "e.g." - a Latin phrase meaning "exempli gratia".

Reply:
I fine tuned my comment a little bit. Thank you.